### PR TITLE
Minor fixes to code

### PR DESCRIPTION
**Pull request by PotatoParser**
Fixes to issues seen on https://lgtm.com/projects/g/aaronjanse/3mux/

### CODE REVIEW

File: ecma48/parser.go [hunk 1]
```diff
@@ -75,7 +75,9 @@ func (p *Parser) Parse(input *bufio.Reader, output chan<- Output) error {
 	rand.Seed(time.Now().Unix())
 
 	go func() {
-		defer recover()
+		defer func() {
+			recover()
+		}()
```
Comment (aaronjanse) [inline ecma48/parser.go:80]: Huh, I thought these are identical?

File: serve.go [hunk 2]
```diff
@@ -222,6 +222,9 @@ func parseFdConn(conn *net.UnixConn) (stdinFd, stdoutFd int) {
 		panic(err)
 	}
 	msgs, err := syscall.ParseSocketControlMessage(buf)
+	if err != nil {
+		panic(err)
+	}
```
Comment (aaronjanse) [inline serve.go:227]: Nice catch!

File: ecma48/parser.go (hunk 1 above)
Comment (PotatoParser) [reply ecma48/parser.go:80]: https://lgtm.com/rules/1511884018877/ My the looks of it, it seems that `defer recover()` doesn't catch panics, but wrapping it into a function does

Comment (aaronjanse) [reply ecma48/parser.go:80]: Oh wow, that's crazy. Thanks for the heads up!

**Pull request merged**